### PR TITLE
feature FINISH-753 Build gem engine

### PR DESCRIPTION
**Pull request by ux**
https://fuseuniversal.atlassian.net/browse/FINISH-753

### CODE REVIEW

**Review by Gikls — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,8 @@
 /pkg/
 /spec/reports/
 /tmp/
+/spec/debug.log
+/Gemfile.lock
```
Comment (bazay) [inline .gitignore:10]: Adding this may fix our issues with rubocop on CircleCI, actually. 
It would be worth adding a separate PR after this one to try and re-add rubocop check

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
+begin
+  require 'pry-byebug'
+rescue LoadError
+end
+
+Dir['./spec/support/**/*.rb'].each { |file| require file }
+
 RSpec.configure do |config|
-  # Disable RSpec exposing methods globally on `Module` and `main`
+  config.example_status_persistence_file_path = '.rspec_status'
```
Comment (bazay) [inline spec/spec_helper.rb]: I don't think we need this and it creates extra trash, but I'll leave up to you.

**Review by bazay — APPROVED** [review]
Looks good. Nice tests too.
Great work 👍
